src/fides/api/db/samples.py

```diff
@@ -106,17 +108,41 @@ def load_sample_connections_from_project() -> List[SampleConnection]:
             )
 
     # Exclude any connections whose "secrets" dict has empty values
+    log.debug(
+        f"Found {len(sample_connections)} sample connections. Skipping any sample connections without configured secrets..."
+    )
     valid_connections = []
     for connection in sample_connections:
-        # If there are no secrets at all, skip this connection
+        # If there are no secrets at all, skip this connection. This shouldn't happen unless sample_connections.yml is misconfigured
         if not connection.secrets:
+            log.debug(
+                f"Skipping sample connection {connection.key}! No secrets exist, cannot create"
+            )
+            continue
+
+        # Check if all secret values exist and are non-empty
+        expected_keys = connection.secrets.keys()  # type: ignore
+        missing_keys = [
+            key
+            for key, value in connection.secrets.items()  # type: ignore
+            if value is None or value == ""
+        ]
+        if len(missing_keys) > 0:
+            log.debug(
+                f"Skipping sample connection {connection.key}. To include this sample, configure ENV secrets for these missing keys: {missing_keys} (see {sample_connections_path}/* for expected ENV vars)"
+            )
             continue
 
-        # Check if all secret values are present and non-empty
-        if all(value and value != "" for value in connection.secrets.values()):  # type: ignore
-            valid_connections.append(connection)
```

Comment:
This was the bug: when `value == False` intentionally, this doesn't work, because `all([False, "localhost", "fides", ...])` will be false.

When a secret value isn't set in the ENV, it'll be `None` (or maybe `""`), so this check needed to be fixed. Whoops